app/services/mapping/goodsshipment/ConsigneeBuilder.scala

```diff
@@ -16,11 +16,22 @@
 
 package services.mapping.goodsshipment
 import forms.declaration.{ConsigneeDetails, EntityDetails}
+import javax.inject.Inject
 import services.Countries.allCountries
+import services.mapping.goodsshipment.ConsigneeBuilder.createConsignee
 import uk.gov.hmrc.http.cache.client.CacheMap
 import wco.datamodel.wco.dec_dms._2.Declaration.GoodsShipment
 import wco.datamodel.wco.declaration_ds.dms._2._
 
+class ConsigneeBuilder @Inject()() {
```

Comment:
so as I see in other comments that Builders should extends trait, trait is missing here

Reply:
Rather than not use the trait, we should fix the trait. If it needs two generic params then so be it :)

Traits make it easier to refactor later, if we want to change the signature of `buildThenAdd` or change the name of the method, we can do it in one keyboard command

Reply:
Shouldn't we use the YAGNI (You ain't going to need it) and KISS principle?
We should also not tangle ourselves with inheritance.

Reply:
Have modified the trait to take 2 generic params